### PR TITLE
added mitm service and restructured arp poisoner and tunnler. added d…

### PR DESCRIPTION
**Pull request by jfeiglin**
…ns log rule. fixed bug in arp_poison.py's get_mac function

### CODE REVIEW

File: mitm_service/arp_poison/arp_poison.py [hunk 1]
```diff
@@ -54,17 +54,23 @@ def _reset_arp_cache(target_ip, target_mac, source_ip, source_mac):
...
         :param target_ip: will intercept requests from this IP
         :param gateway_ip: will intercept responses from this IP (directed at :param target_ip)
         :param interval: how often to run arp poison
         """
         self.target_ip = target_ip
-        self.target_mac = MITMService._get_mac(self.target_ip, mac_resolve_max_tries)
+        if target_mac is None:
+            self.target_mac = ARPPoisonService.get_mac(self.target_ip, mac_resolve_max_tries)
+        else:
+            self.target_mac = target_mac
```
Comment (benjaminbaker-dev) [inline mitm_service/arp_poison/arp_poison.py]: self.target_mac = target_mac or type(self).get_mac() is much cleaner

File: mitm_service/arp_poison/arp_poison.py [hunk 2]
```diff
@@ -75,16 +81,16 @@ def _run_mitm(self):
         inserts this machine between target_ip and gateway_ip
         """
         while self._should_spoof:
-            MITMService._poison_arp_cache(self.target_ip, self.target_mac, self.gateway_ip)
-            MITMService._poison_arp_cache(self.gateway_ip, self.gateway_mac, self.target_ip)
+            ARPPoisonService._poison_arp_cache(self.target_ip, self.target_mac, self.gateway_ip)
+            ARPPoisonService._poison_arp_cache(self.gateway_ip, self.gateway_mac, self.target_ip)
```
Comment (benjaminbaker-dev) [inline mitm_service/arp_poison/arp_poison.py]: type(self)

File: mitm_service/arp_poison/arp_poison.py [hunk 3]
```diff
@@ -75,16 +81,16 @@ def _run_mitm(self):
...
             sleep(self.interval)
 
     def _restore_normal_arp(self):
         """
         resets ip/mac mapping so that gateway ip resolves to gateway mac, and target ip resolves to target mac
         """
-        MITMService._reset_arp_cache(self.target_ip, self.target_mac, self.gateway_ip, self.gateway_mac)
-        MITMService._reset_arp_cache(self.gateway_ip, self.gateway_mac, self.target_ip, self.target_mac)
+        ARPPoisonService._reset_arp_cache(self.target_ip, self.target_mac, self.gateway_ip, self.gateway_mac)
+        ARPPoisonService._reset_arp_cache(self.gateway_ip, self.gateway_mac, self.target_ip, self.target_mac)
```
Comment (benjaminbaker-dev) [inline mitm_service/arp_poison/arp_poison.py]: type(self)

File: mitm_service/mitm_service.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
+import binascii
+import socket
+import struct
+import fcntl
+
+from mitm_service.arp_poison.arp_poison import ARPPoisonService
+from mitm_service.tunneler.tunneler import L2Tunnel
+
+SIOCGIFHWADDR = 0x8927
```
Comment (benjaminbaker-dev) [inline mitm_service/mitm_service.py:10]: ???

File: mitm_service/mitm_service.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        self.my_mac = get_interface_mac(self.interface)
+
+        if target_mac is None:
+            self.target_mac = ARPPoisonService.get_mac(self.target_ip, 5)
+        else:
+            self.target_mac = target_mac
+        if gateway_mac is None:
+            self.gateway_mac = ARPPoisonService.get_mac(self.gateway_ip, 5)
+        else:
+            self.gateway_mac = gateway_mac
```
Comment (benjaminbaker-dev) [inline mitm_service/mitm_service.py]: self.target_mac = target_mac or type(self).get_mac() is much cleaner

File: mitm_service/tunneler/tunneler.py [hunk 6]
```diff
@@ -155,7 +155,11 @@ def forward_loop(self):
             if recv_ipheader.dst_ip != self.target_ip and recv_ipheader.src_ip != self.target_ip:
                 continue
 
-            disrupted_packet = self.disrupt_layers(data)
+            try:
+                disrupted_packet = self.disrupt_layers(data)
+            except DropPacketException:
+                # some filter in disrupt_layers raised a drop packet exception, so drop this packet
+                continue
```
Comment (benjaminbaker-dev) [inline mitm_service/tunneler/tunneler.py:161]: write log ?

**Review by benjaminbaker-dev — COMMENTED** [review]
small changes, looks great

**Review by benjaminbaker-dev — CHANGES_REQUESTED** [review]
requesting changes lol

File: mitm_service/mitm_service.py (hunk 4 above)
Comment (jfeiglin) [reply mitm_service/mitm_service.py:10]: linux ioctl constant, thats actually what its called. Look it up.

File: mitm_service/mitm_service.py (hunk 5 above)
Comment (jfeiglin) [reply mitm_service/mitm_service.py]: type(self) does not have a get_mac function, so it'd have to stay as ARPPoisonService.get_mac

File: mitm_service/mitm_service.py (hunk 4 above)
Comment (benjaminbaker-dev) [reply mitm_service/mitm_service.py:10]: lol pls add "linux ioctl constant" this as comment

File: mitm_service/tunneler/tunneler.py (hunk 6 above)
Comment (jfeiglin) [reply mitm_service/tunneler/tunneler.py:161]: No need, if this exception is raised it means a user-supplied filter raised it on purpose, so they can print a log if they want.

**Review by benjaminbaker-dev — APPROVED** [review]

**Pull request merged**